puppet/modules/rbenv/manifests/build.pp

```diff
@@ -0,0 +1,90 @@
+# Calling this define will install Ruby in your default rbenv
+# installs directory. Additionally, it can define the installed
+# ruby as the global interpretter. It will install the bundler gem.
+#
+# @param install_dir
+#   This is set when you declare the rbenv class. There is no
+#   need to overrite it when calling the rbenv::build define.
+#
+# @param global
+#   This is used to set the ruby to be the global interpreter.
+#
+# @param env
+#   This is used to set environment variables when compiling ruby.
+#
+# @param rubygems_version
+#  This is used to set a specific version of rubygems.
+#
+# @param bundler_version
+#   This is used to set a specific version of bundler.
+#
+# @example Install Ruby 2.7.8 as global
+#   rbenv::build { '2.7.8':
+#     global => true,
+#   }
+#
+define rbenv::build (
```

Comment:
I must admit I find the name of this define odd.
Yes, it's using `ruby-build` under the hood, but the `rbenv` command is `rbenv install` and to be even more precise, I'd probably not use the "action" (build/install) but the "object" (ruby) that we're managing here and call it `rbenv::ruby`.

That all said, this is based on the work in jdowning/rbenv and I have no idea how much we want to diverge from it, naming wise.

Reply:
No real preference from me